### ISSUE TITLE
Windows Cairo error when trying to export to pdf

### ISSUE THREAD

**Issue by senpaiinduhsheet** [xournalpp/xournalpp#4256]
(Please complete the following information, and then delete this line)

On windows 10 Xournal version 1.1.1

Unable to export because of cairo error:
![image](https://user-images.githubusercontent.com/83674373/192672571-01f0861a-2171-4015-bbc6-0c012696c271.png)

Also when trying to annotate on a preexisting pdf I get this sorry if this is just a user error new to the app want to get off of onenote
![image](https://user-images.githubusercontent.com/83674373/192672385-f400abcc-4298-4f0d-b85d-201d457916ce.png)

**To Reproduce**
Steps to reproduce the behavior of the cairo error:
1. Open xournal
2. sketch anything
3. export to pdf
4. error comes

Steps to reproduce the behavior of the annotate issue:
1. In files "annotate pdf"
2. write on the pdf I do it anywhere
3. attempt to save 
4. error

**Expected behavior**
I thought I would just be able to save them as a pdf I have a pdf app called "PDF24"

**Comment by Technius** [comment]
The cairo error looks concerning to me. Does this happen when you try to save the PDF to specific places, or anywhere?

**Comment by senpaiinduhsheet** [comment]
It didn't really matter where I saved it would still give me that error. However, I think it was because I was writing on the background layer and trying to save it. I assumed I could open the app and just use like OneNote. If you think it would be a bigger issue that I'm understanding I can give you more info on what I was doing before. Currently I can save and annotate on PDFs as well as export them.

**Comment by Technius** [comment]
To double check that I understand the issue correctly:

>  Steps to reproduce the behavior of the cairo error

Is this from a complete new, blank document, or after opening up a PDF page? As far as I'm aware, if you try to draw a stroke on the "Background" layer, the application will automatically create a new layer for you.

> Steps to reproduce the behavior of the annotate issue:

This message shows if you try to export directly back to the original PDF. For example, if you use "Annotate PDF" to load "example.pdf", and then you use "Export as PDF" to save back to "example.pdf", Xournal++ will currently not allow you to do that. The reason is that overwriting the background PDF can cause confusion if you are working with an `.xopp` file.

A workaround is to name the exported file something else instead, like "example2.pdf" (note that after exporting, your annotations will be baked into the file and cannot be edited).

See #2363 for further discussion on overwriting background PDFs.

**Comment by senpaiinduhsheet** [comment]
Yes it would either be a freshly opened PDF or when I first open Xournal. I don't think it does automatically add layers for me I usually add the layers myself. I only get the error when I attempt to export the file in PDF format.

**Comment by senpaiinduhsheet** [comment]
Apparently I'm getting the errors again I don't know what caused them to not export but same issue. 
Open New or Existing .xopp file attempt to export Cairo error.
